### DISCUSSION TITLE
How to build artifacts? - Installation instructions missing

### DISCUSSION THREAD

**Discussion by marcospb19** [SeaQL/FireDBG.for.Rust#4] [Q&A] [answered]
Hi! At the time of writing, it's not clear from the README.md how can I build FireDBG artifacts.

Can someone give me a light on this?

**Comment by tyt2y3** [comment]
In summary we bundle [codelldb](https://github.com/vadimcn/codelldb)'s binaries with our binaries. We'll try to release a script for that process.

**Comment by arduano** [comment]
Having a working build script as reference would be pretty useful, as I'm on one of the unsupported platforms (Arch linux), and was attempting to get it compiling myself. I don't mind getting my hands dirty, but yeah having a reference would be nice.

Maybe even a github workflow for automated builds sometime?

**Comment by clemarescx** [comment]
I too would like to see releases for Arch-based distros (Manjaro user here).
For the install script, to get _libc++ ABI_ this should work for [Arch-based distributions using Pacman](https://en.wikipedia.org/wiki/Category:Pacman-based_Linux_distributions) (Garuda Linux and SteamOS also use it, but somehow are not listed):
```shell
check_pacman_install libc++abi

check_pacman_install() {
    if [[ ! $(pacman -Qs $1) ]]; then
        sudo pacman -S --noconfirm $1
    fi
}
```

**Comment by Speykious** [comment]
Oh well, I was gonna PR a change to the install script to add support for Arch, until I realized the provided precompiled binaries are separated per Linux distro.
From the install script, it seems that a different version of `libc++abi` is installed on each distro. Is there no way to just have one Linux release, or will it have to have at least one per `libc++abi` version?
If anything it could have an appimage.

**Reply by tyt2y3** [reply]
Yes, that's a bit of bummer. The problem is linking with libc, which is different on each distro, and we want to play safe, so we try to compile on each platform.

**Comment by connortsui20** [comment]
I was able to get Arch working by changing the install script [slightly](https://github.com/SeaQL/FireDBG.for.Rust/discussions/9)

**Comment by billy1624** [comment, marked as answer]
Hi all, thanks for the interest!!

We just released the Dockerfile and a guide for building FireDBG from source:
- https://github.com/SeaQL/FireDBG.for.Rust/tree/main/build
- https://github.com/SeaQL/FireDBG.for.Rust/blob/main/BUILDING.md
